### PR TITLE
Feature/ingroup azureapplication capacity

### PR DESCRIPTION
**Pull request by ladlavj**
Supports the case where the Azure application only way to use Power BI APIs and where the user is MFA.
Features.

1. New provider parameter "grant_type=client_credentials" to support application only authorization.
2. New workspace resource parameter to assign it to dedicated capacity.
3. PBIX import, Datasource and report API URLs changed with "/groups/<group_id>"
4. New refresh schedule resource parameter "workspace_id"

Workspace id is a must when resources are created by the user, who is not Power BI service admin and just 'Capacity Admin'.

### CODE REVIEW

File: internal/powerbi/resource_refresh_schedule_test.go [hunk 1]
```diff
@@ -129,11 +132,11 @@ func TestAccRefreshSchedule_basic(t *testing.T) {
...
 			// final step checks importing the current state we reached in the step above
-			{
-				ResourceName:      "powerbi_refresh_schedule.test",
-				ImportState:       true,
-				ImportStateVerify: true,
-			},
+			// {
+			// 	ResourceName:      "powerbi_refresh_schedule.test",
+			// 	ImportState:       true,
+			// 	ImportStateVerify: false,
```
Comment (alex-davies) [inline internal/powerbi/resource_refresh_schedule_test.go:138]: Commented out test

This is also highlights a concern with adding another required identifier field, im not sure what the impacts on importing become

File: internal/powerbi/resource_workspace_test.go [hunk 2]
```diff
@@ -31,19 +31,19 @@ func TestAccWorkspace_basic(t *testing.T) {
...
-					name = "Acceptance Test Workspace %s - Updated"
-				}
-				`, workspaceSuffix),
-				Check: resource.ComposeTestCheckFunc(
-					testCheckWorkspaceExistsWithName("powerbi_workspace.test", fmt.Sprintf("Acceptance Test Workspace %s - Updated", workspaceSuffix)),
-					resource.TestCheckResourceAttrSet("powerbi_workspace.test", "id"),
-					resource.TestCheckResourceAttr("powerbi_workspace.test", "name", fmt.Sprintf("Acceptance Test Workspace %s - Updated", workspaceSuffix)),
-				),
-			},
+			// second step Assigns capacity to workspace
```
Comment (alex-davies) [inline internal/powerbi/resource_workspace_test.go:34]: Another commented out test

File: internal/powerbiapi/import.go [hunk 3]
```diff
@@ -123,13 +123,38 @@ func (client *Client) PostImportInGroup(groupID string, datasetDisplayName strin
 }
 
 // WaitForImportToSucceed waits until the specified import
-func (client *Client) WaitForImportToSucceed(importID string, timeout time.Duration) (*GetImportResponse, error) {
+// func (client *Client) WaitForImportToSucceed(importID string, timeout time.Duration) (*GetImportResponse, error) {
```
Comment (alex-davies) [inline internal/powerbiapi/import.go:126]: This can probably be deleted

**Comment by alex-davies** [comment]
Automated tests are failing, but it appears its because of the use of secrets. Do you have a powerbi setup to add secrets to your fork repo so the tests can run (https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow implies that it will use the secrets on your fork when running tests)

**Comment by alex-davies** [comment]
Unknowing that you were actively looking at this I had done some work last weekend as well to support client_auth. There is lots of overlap with what you have done with some minor differences in some of the smaller details (such as specify grant_type parameter vs determining if username/password is present). 

There were a few outstanding pieces which both our implementation is lacking
* Some more detailed docs on this authentication mode and how it is configured in powerbi. Ideally in the authentication guide
* Understanding how an additional key in workspace impacts imports
  * It maybe we disable imports on some resources, which for pbix makes sense as there are lots of fields we cant import anyway on them 
* How to update automated tests to include both authentication modes

**Comment by alex-davies** [comment]
You also have some really good enhancements bundled in here, in particular the capacity_id and worksapce permissions. Is it possible you could break those out into separate PRs. Those changes are less fundamental and easier to discuss in isolation

**Comment by ladlavj** [comment]
Closing this PR and raising separate PRs for each feature tracked under #14.

**Pull request closed without merging**